data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/DeserializeBlocksTest.java

```diff
@@ -16,70 +16,159 @@
+import tech.pegasys.teku.spec.datastructures.blocks.SignedBlockContainer;
+import tech.pegasys.teku.spec.datastructures.blocks.versions.deneb.SignedBlindedBlockContents;
+import tech.pegasys.teku.spec.datastructures.blocks.versions.deneb.SignedBlindedBlockContentsSchema;
+import tech.pegasys.teku.spec.datastructures.blocks.versions.deneb.SignedBlockContents;
+import tech.pegasys.teku.spec.datastructures.blocks.versions.deneb.SignedBlockContentsSchema;
+import tech.pegasys.teku.spec.util.DataStructureUtil;
 
 public class DeserializeBlocksTest {
 
-  public static final DeserializableTypeDefinition<SignedBeaconBlock> SIGNED_BEACON_BLOCK_TYPE =
-      DeserializableTypeDefinition.object(SignedBeaconBlock.class)
-          .name("SignedBeaconBlock")
-          .initializer(SignedBeaconBlock::new)
-          .build();
-  public static final DeserializableTypeDefinition<SignedBlockContents> SIGNED_BLOCK_CONTENTS_TYPE =
-      DeserializableTypeDefinition.object(SignedBlockContents.class)
-          .name("SignedBlockContents")
-          .initializer(SignedBlockContents::new)
-          .build();
-
-  public static final DeserializableOneOfTypeDefinition<BlockContainer, BlockContainerBuilder>
-      DESERIALIZABLE_ONE_OF_TYPE_DEFINITION =
+  static Spec spec = TestSpecFactory.createMinimalDeneb();
+  DataStructureUtil denebData = new DataStructureUtil(spec);
+  public static final DeserializableOneOfTypeDefinition<SignedBlockContainer, BlockContainerBuilder>
+      DESERIALIZABLE_ONE_OF_SIGNED_BEACON_BLOCK_OR_SIGNED_BLOCK_CONTENTS =
           DeserializableOneOfTypeDefinition.object(
-                  BlockContainer.class, BlockContainerBuilder.class)
-              .description(
-                  "Submit a signed beacon block to the beacon node to be imported."
-                      + " The beacon node performs the required validation.")
+                  SignedBlockContainer.class, BlockContainerBuilder.class)
               .withType(
-                  SignedBeaconBlock.isInstance,
+                  SignedBeaconBlock.isSignedBeaconBlockInstance,
                   s -> !s.contains("blob_sidecars"),
-                  SIGNED_BEACON_BLOCK_TYPE)
+                  spec.getGenesisSchemaDefinitions()
+                      .getSignedBeaconBlockSchema()
+                      .getJsonTypeDefinition())
               .withType(
-                  SignedBlockContents.isInstance,
+                  SignedBlockContents.isSignedBlockContentsInstance,
                   s -> s.contains("blob_sidecars"),
-                  SIGNED_BLOCK_CONTENTS_TYPE)
+                  SignedBlockContentsSchema.create(
+                          spec.getGenesisSpecConfig().toVersionDeneb().orElseThrow(),
+                          SignedBlobSidecarSchema.create(
+                              BlobSidecarSchema.create(
+                                  new BlobSchema(
+                                      spec.getGenesisSpecConfig().toVersionDeneb().orElseThrow()))),
```

Comment:
Basically this, it's just generally not fully exposed here...
if you look at `GetLightClientBootstrap` as an example, but basically you'd be doing the .create elsewhere, and here it'd be something like
```
SchemaDefinitionsDeneb.required(
                schemaDefinitionCache.getSchemaDefinition(SpecMilestone.DENEB))
            .getSignedBlockContentsSchema()
            .getJsonTypeDefinition()
```

Reply:
admittedly arugable either way in an actual test, but in the endpoints it'd look like this...